pkg/ddl/ddl_worker.go

```diff
@@ -899,7 +907,27 @@ func (w *worker) prepareTxn(job *model.Job) (kv.Transaction, error) {
 	return txn, err
 }
 
-func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) (int64, error) {
+// runOneJobStep runs one step of the DDL job and persist the states change. One
+// *step* is defined as the following reason:
+//
+// - TiDB uses "Asynchronous Schema Change in F1", one job may have multiple
+// *steps* each for a "schema state" such as 'delete only' -> 'write only'. The
+// cluster can only have at most two adjacent schema states for a DDL object.
+// This function forwards one (schema state) *step* and caller
+// `runOneJobStepAndWaitSync` waits for other nodes to catch up.
+//
+// - In order to make sure every job is deterministic, idempotent to provide
+// correctness through failover, this function will decide and persist the
+// arguments of a job as a separate *step*. These steps will reuse "schema state"
+// changes, see onRecoverTable as an example.
```

Comment:
you include done->sync in this part? it's not for failover, it must be done in a separate step after wait version.

Reply:
No "done->sync" is included in the item of your above comment. 

This part is like the reason in the comments of onRecoverTable. If the job will change system states (like disable GC) and revert it afterward, if must save the GC state before get running. Otherwise if the job fails between changing states and reverting states, we can't recover the original states.